bagitobjecttransfer/Containerfile

```diff
@@ -12,6 +12,13 @@ COPY . .
 # Make directories, install dependencies and prepare entrypoint script
 RUN pip install --upgrade setuptools wheel \
     && pip install -r requirements.txt \
+    && apt-get update \
+    && apt-get install -y curl \
+    && curl -fsSL https://deb.nodesource.com/setup_22.x -o nodesource_setup.sh \
+    && bash nodesource_setup.sh \
+    && apt-get install -y nodejs \
+    && npm install -g npm \
+    && cd /app && npm install \
```

Comment:
Is the npm install step working now? You mentioned before that `npm install` wasn't actually installing the yuglify binary. If this is fixed, disregard this comment.

Reply:
Strangely, `npm install` runs on the prod container, but not the dev:

On the prod container:
```shell
╭─    ~/De/Secure-Record-Transfer/bagitobjecttransfer    minify-css-and-js-during-build !1 
╰─ DOCKER_DEFAULT_PLATFORM=linux/arm64 docker compose --env-file .prod.env -f compose-arm.prod.yml up -d --build --remove-orphans
WARN[0000] /Users/X1Carbon/Desktop/Secure-Record-Transfer/bagitobjecttransfer/compose-arm.prod.yml: `version` is obsolete
[+] Building 26.2s (15/15) FINISHED                        docker:desktop-linux
 => [rq internal] load build definition from Containerfile                 0.0s
 => => transferring dockerfile: 771B                                       0.0s
 => [app internal] load metadata for docker.io/library/python:3.10         0.6s
 => [rq internal] load .dockerignore                                       0.0s
 => => transferring context: 237B                                          0.0s
 => [app 1/4] FROM docker.io/library/python:3.10@sha256:428430918ab721099  0.0s
 => => resolve docker.io/library/python:3.10@sha256:428430918ab72109921ef  0.0s
 => [rq internal] load build context                                       0.0s
 => => transferring context: 132.35kB                                      0.0s
 => CACHED [app 2/4] WORKDIR /app                                          0.0s
 => [rq 3/4] COPY . .                                                      0.1s
 => [rq 4/4] RUN pip install --upgrade setuptools wheel     && pip insta  24.0s
 => [rq] exporting to image                                                1.1s
 => => exporting layers                                                    1.1s
 => => writing image sha256:7ec1180d5401ed0f4f55c6a26027efd4992e7ab20a738  0.0s
 => => naming to docker.io/library/bagitobjecttransfer-rq                  0.0s
 => [app internal] load build definition from Containerfile                0.0s
 => => transferring dockerfile: 771B                                       0.0s
 => [app internal] load .dockerignore                                      0.0s
 => => transferring context: 237B                                          0.0s
 => [app internal] load build context                                      0.0s
 => => transferring context: 73.28kB                                       0.0s
 => CACHED [app 3/4] COPY . .                                              0.0s
 => CACHED [app 4/4] RUN pip install --upgrade setuptools wheel     && pi  0.0s
 => [app] exporting to image                                               0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:c54a7aa5d44310f5d3bdf0f3afbbbd3a35381ff50bbdb  0.0s
 => => naming to docker.io/library/bagitobjecttransfer-app                 0.0s
[+] Running 7/7
 ✔ Container recordtransfer_mailhog        Removed                         0.1s
 ✔ Container bagitobjecttransfer-clamav-1  Running                         0.0s
 ✔ Container bagitobjecttransfer-db-1      Running                         0.0s
 ✔ Container bagitobjecttransfer-redis-1   Running                         0.0s
 ✔ Container bagitobjecttransfer-rq-1      Started                         1.3s
 ✔ Container bagitobjecttransfer-app-1     Started                         1.0s
 ✔ Container bagitobjecttransfer-nginx-1   Started                         0.7s

╭─    ~/De/Secure-Record-Transfer/bagitobjecttransfer    minify-css-and-js-during-build !1 
╰─ docker exec -it  Container bagitobjecttransfer-app-1 /bin/bash

What's next:
    Try Docker Debug for seamless, persistent debugging tools in any container or image → docker debug Container
    Learn more at https://docs.docker.com/go/debug-cli/
Error response from daemon: No such container: Container

╭─    ~/De/Secure-Record-Transfer/bagitobjecttransfer    minify-css-and-js-during-build !1 
╰─ docker exec -it bagitobjecttransfer-app-1 /bin/bash
root@f936576a3f10:/app# ls
Containerfile	      compose.prod.yml	   package.json
bagitobjecttransfer   docker		   recordtransfer
caais		      manage.py		   requirements.dev.txt
clamav		      media		   requirements.txt
compose-arm.dev.yml   node_modules	   static
compose-arm.prod.yml  nodesource_setup.sh  templates
compose.dev.yml       package-lock.json
root@f936576a3f10:/app# exit
exit
```

On dev container, note how `node_modules` is missing.
``` shell
╭─    ~/De/Secure-Record-Transfer/bagitobjecttransfer    minify-css-and-js-during-build !1 
╰─ DOCKER_DEFAULT_PLATFORM=linux/arm64 docker compose --env-file .dev.env -f compose-arm.dev.yml up -d --build --remove-orphans
[+] Building 25.5s (15/15) FINISHED                        docker:desktop-linux
 => [rq internal] load build definition from Containerfile                 0.0s
 => => transferring dockerfile: 771B                                       0.0s
 => [app internal] load metadata for docker.io/library/python:3.10         0.8s
 => [rq internal] load .dockerignore                                       0.0s
 => => transferring context: 237B                                          0.0s
 => [app 1/4] FROM docker.io/library/python:3.10@sha256:428430918ab721099  0.0s
 => => resolve docker.io/library/python:3.10@sha256:428430918ab72109921ef  0.0s
 => [rq internal] load build context                                       0.0s
 => => transferring context: 156.58kB                                      0.0s
 => CACHED [app 2/4] WORKDIR /app                                          0.0s
 => [rq 3/4] COPY . .                                                      0.1s
 => [rq 4/4] RUN pip install --upgrade setuptools wheel     && pip insta  23.0s
 => [rq] exporting to image                                                1.3s
 => => exporting layers                                                    1.3s
 => => writing image sha256:c5d02402c550f14bb5e8e658eacc7a11671c1398ee391  0.0s
 => => naming to docker.io/library/bagitobjecttransfer-rq                  0.0s
 => [app internal] load build definition from Containerfile                0.0s
 => => transferring dockerfile: 771B                                       0.0s
 => [app internal] load .dockerignore                                      0.0s
 => => transferring context: 237B                                          0.0s
 => [app internal] load build context                                      0.0s
 => => transferring context: 73.28kB                                       0.0s
 => CACHED [app 3/4] COPY . .                                              0.0s
 => CACHED [app 4/4] RUN pip install --upgrade setuptools wheel     && pi  0.0s
 => [app] exporting to image                                               0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:20434019da70e6a12b5b25f7033fb0ef1899f2db65ee1  0.0s
 => => naming to docker.io/library/bagitobjecttransfer-app                 0.0s
[+] Running 11/11
 ✔ Container bagitobjecttransfer-db-1      Removed                         0.9s
 ✔ Container bagitobjecttransfer-nginx-1   Removed                         0.2s
 ✔ Container bagitobjecttransfer-redis-1   Recreated                       0.4s
 ✔ Container bagitobjecttransfer-clamav-1  Recreated                       0.3s
 ✔ Container recordtransfer_mailhog        Started                         1.5s
 ✔ Container bagitobjecttransfer-rq-1      Recreated                       0.4s
 ✔ Container bagitobjecttransfer-app-1     Recreated                       0.4s
 ✔ Container recordtransfer_clamav         Started                         0.4s
 ✔ Container redis_server                  Sta...                          0.4s
 ✔ Container recordtransfer_rq_workers     Started                         0.2s
 ✔ Container recordtransfer_django_app     Started                         0.2s

╭─    ~/De/Secure-Record-Transfer/bagitobjecttransfer    minify-css-and-js-during-build !1 
╰─ docker exec -it recordtransfer_django_app /bin/bash
root@fd556bb46f2a:/app# ls
Containerfile	      compose.prod.yml		    package.json
bagitobjecttransfer   development_database.sqlite3  recordtransfer
caais		      docker			    requirements.dev.txt
clamav		      example.dev.env		    requirements.txt
compose-arm.dev.yml   example.prod.env		    static
compose-arm.prod.yml  manage.py			    templates
compose.dev.yml       media
```

The app itself on dev has no errors, and serves static assets fine.

Reply:
That's probably because of the volume mount that caused the issues with the entrypoint script. Since we ignore node_modules/ in the .dockerignore I wonder if that makes it so that that folder can't be created (else that folder would show up locally).

Since it's serving static assets fine in dev, even though the executable isn't there, I'd say it's OK to be merged like this.